cognite/neat/_data_model/validation/dms/_limits_check.py

```diff
@@ -0,0 +1,377 @@
+          * Data type (Int32, Int64, DirectRelation, etc.)
+          * Presence of btree index
+          * Default vs maximum limits
+
+        Enum properties are skipped (have separate 32-value limit).
+
+        In additive mode, counts include properties from both local and CDF
+        versions of the container.
+
+        Returns:
+            List of ConsistencyError issues, one per limit violation found.
+            Empty list if all containers are within limits.
+        """
+
+        errors: list[ConsistencyError] = []
+
+        merged_containers = self.merged_containers
+
+        for container_ref in self.local_resources.containers_by_reference.keys():
+            container = merged_containers.get(container_ref)
+            if not container:
+                raise RuntimeError(f"Container {container_ref!s} not found in merged containers. This is a bug!")
+
+            if container.properties:
+                properties_by_index_type = self.container_property_by_index_type(container)
+
+                if len(container.properties) > DMSDefaultLimits.container.properties_per_container:
+                    errors.append(
+                        ConsistencyError(
+                            message=(
+                                f"Container {container_ref!s} has {len(container.properties)} properties, "
+                                "which exceeds the limit of "
+                                f"{DMSDefaultLimits.container.properties_per_container} properties per container."
+                            ),
+                            code=self.code,
+                        )
+                    )
+
+                for property_id, property_ in container.properties.items():
+                    type_ = property_.type
+
+                    if isinstance(type_, EnumProperty):
+                        continue
+
+                    if type_.list and hasattr(type_, "max_list_size") and type_.max_list_size:
+                        has_btree_index = (
+                            property_id in properties_by_index_type[BtreeIndex.model_fields["index_type"].default]
+                        )
+                        max_allowed_limit = DMSDefaultLimits.container.get_limit_for_data_type(type_, has_btree_index)
+                        if type_.max_list_size > max_allowed_limit:
+                            errors.append(
+                                ConsistencyError(
+                                    message=(
+                                        f"Container {container_ref!s} has property {property_id} with list size "
+                                        f"{type_.max_list_size}, which exceeds the limit of {max_allowed_limit} "
+                                        f"for data type {type_.__class__.__name__}."
+                                    ),
+                                    code=self.code,
+                                )
+                            )
```

Comment:
Suggest using `ListablePropertyTypeDefinition` instead:

```suggestion
                    if not isinstance(type_, ListablePropertyTypeDefinition) or type_.max_list_size is None:
                        continue

                    has_btree_index = (
                        property_id in properties_by_index_type[BtreeIndex.model_fields["index_type"].default]
                    )
                    max_allowed_limit = DMSDefaultLimits.container.get_limit_for_data_type(type_, has_btree_index)
                    if type_.max_list_size > max_allowed_limit:
                        errors.append(
                            ConsistencyError(
                                message=(
                                    f"Container {container_ref!s} has property {property_id} with list size "
                                    f"{type_.max_list_size}, which exceeds the limit of {max_allowed_limit} "
                                    f"for data type {type_.__class__.__name__}."
                                ),
                                code=self.code,
                            )
                        )
```

Reply:
more readable